File: Binary-Search/binary_search.test.js
const binarySearch = require("./binary_search")

const arr = ["Egor", "David", "Zack", "Alex", "Job", "Darell"]
arr.sort()

test("Find index of the last element in array, equals = 5",  () => {
    expect(binarySearch(arr, "Zack")).toBe(5)
})

test("Find index of the first element in array, equals = 0",  () => {
    expect(binarySearch(arr, "Alex")).toBe(0)
})

test("Find non-existent element in array, equals = -1",  () => {
    expect(binarySearch(arr, "hello")).toBe(-1)
})

File: binary_search/binary_search_test.go
package binary_search

import (
	"sort"
	"testing"

	"github.com/stretchr/testify/assert"
)

var arr = []string{"Egor", "David", "Zack", "Alex", "Job", "Darell"}

func init() {
	sort.Strings(arr)
}

func TestBinarySearch_FindLastElement(t *testing.T) {
	idx := BinarySearchStr(arr, "Alex")
	assert.Equal(t, 0, idx)
}

func TestBinarySearch_FindFirstElement(t *testing.T) {
	idx := BinarySearchStr(arr, "Zack")
	assert.Equal(t, len(arr)-1, idx)
}

func TestBinarySearch_EmptyResult(t *testing.T) {
	idx := BinarySearchStr(arr, "hello")
	assert.Equal(t, -1, idx)
}
